src/server/session.ts

```diff
@@ -26,14 +26,7 @@ namespace ts.server {
     }
 
     function shouldSkipSematicCheck(project: Project) {
-        if (project.getCompilerOptions().skipLibCheck !== undefined) {
-            return false;
-        }
-
-        if ((project.projectKind === ProjectKind.Inferred || project.projectKind === ProjectKind.External) && project.isJsOnlyProject()) {
-            return true;
-        }
-        return false;
+        return (project.projectKind === ProjectKind.Inferred || project.projectKind === ProjectKind.External) && project.isJsOnlyProject();

```

Comment:
correct. if you have configured project the check is `false`, and the normal checking for semantic errors proceed. this is only if you have external or inferred project, with only .js files.